bundles/org.openhab.binding.octopusenergy/src/main/java/org/openhab/binding/octopusenergy/internal/OctopusEnergyHandlerFactory.java

```diff
@@ -0,0 +1,91 @@
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.octopusenergy.internal.handler.OctopusEnergyBridgeHandler;
+import org.openhab.binding.octopusenergy.internal.handler.OctopusEnergyElectricityMeterPointHandler;
+import org.openhab.binding.octopusenergy.internal.handler.OctopusEnergyGasMeterPointHandler;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.binding.BaseThingHandlerFactory;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.thing.binding.ThingHandlerFactory;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OctopusEnergyHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Rene Scherer - Initial contribution
+ */
+@NonNullByDefault
+@Component(configurationPid = "binding.octopusenergy", service = ThingHandlerFactory.class)
+public class OctopusEnergyHandlerFactory extends BaseThingHandlerFactory {
+
+    private final Logger logger = LoggerFactory.getLogger(OctopusEnergyHandlerFactory.class);
+
+    private OctopusEnergyApiHelper apiHelper = new OctopusEnergyApiHelper();
+
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = Stream
+            .of(BRIDGE_THING_TYPES_UIDS, OctopusEnergyBindingConstants.SUPPORTED_THING_TYPES_UIDS)
+            .flatMap(Collection::stream).collect(Collectors.toSet());
+
+    /**
+     * Returns true if the factory supports the given thing type.
+     */
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    /**
+     * Returns a newly created thing handler for the given thing.
+     */
+    @Override
+    protected @Nullable ThingHandler createHandler(Thing thing) {
+        logger.debug("createHandler - create handler for {}", thing);
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+
+        if (thingTypeUID.equals(THING_TYPE_ELECTRICITY_METER_POINT)) {
+            return new OctopusEnergyElectricityMeterPointHandler(thing, apiHelper);
+        } else if (thingTypeUID.equals(THING_TYPE_GAS_METER_POINT)) {
+            return new OctopusEnergyGasMeterPointHandler(thing, apiHelper);
+        } else if (thingTypeUID.equals(THING_TYPE_BRIDGE)) {
+            return new OctopusEnergyBridgeHandler((Bridge) thing, apiHelper);
+        }
+        return null;
+    }
+
+    @Reference
+    protected void setHttpClientFactory(HttpClientFactory httpClientFactory) {
+        apiHelper.setHttpClient(httpClientFactory.getCommonHttpClient());
+    }
```

Comment:
You could inject this via the constructor. Then, the all the Nullables for httpClient can be removed downstream.